content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,285 @@
+
+```javascript
+describe("Generated test cases", () => {
+  const testCases = generateTestCases();
+  testCases.forEach(([msg, input, expected]) => {
+    it(msg, () => {
+      const actual = userSolution(input);
+      assert.strictEqual(actual, expected);
+    });
+  });
+});
+```
+
+This technique is liked by authors familiar with testing frameworks that provide parameterized or generated test cases out of the box, like NUnit or JUnit. However, some caution is needed when this approach is used. Test suites organized like this can become large and can flood the test output panel with many entries, making it unreadable or causing performance problems in client browsers.
+
+### Chai
+
+JavaScript kata should use [Chai](https://www.chaijs.com/) as the assertion library.
+
+:::note
+For simplicity, through the rest of this article it's assumed that the `chai` package is imported and `assert` is extracted from it with the following statement:
+
+```javascript
+const chai = require('chai');
+const assert = chai.assert;
+```
+
+This is a convention used in many JavaScript kata, but it's not a requirement, and authors can choose to import the package in any way they find suitable for them. Also, the `should` or `expect` style can also be used from `chai`.
+:::
+
+The most commonly used assertions include:
+
+- [`assert.strictEqual(actual, expected[, message])`](https://www.chaijs.com/api/assert/#method_strictequal) - Asserts that `actual === expected`.
+- [`assert.deepEqual(actual, expected[, message])`](https://www.chaijs.com/api/assert/#method_deepequal) - Asserts that two arrays or objects have the same values. `strictEqual` does not work since arrays and objects are reference types in JS.
+- [`assert.closeTo(actual, expected, delta[, message])`](https://www.chaijs.com/api/assert/#method_closeto) - Asserts that `abs(actual - expected) <= delta`. Should be used when comparing floating-point numbers.
+
+See the [Chai docs](https://www.chaijs.com/api/assert/) for all the methods.
+
+:::note
+The [`truncateThreshold` option](https://www.chaijs.com/guide/styles/#configtruncatethreshold) controls how arrays are stringified when `expected` and `actual` values are prepared for display in assertion messages. Setting it to 0 provides the most detailed feedback even for large arrays, but then assertion messages can flood the console and make the output difficult to read.
+
+```javascript
+chai.config.truncateThreshold = 0;
+```
+
+:::
+
+### Test feedback
+
+Unlike Python or Ruby's test framework, Mocha handles combines multiple assertions in an `it` block into just one. For example, if all the assertions pass, then it will just show one `Test Passed`, and not a `Test Passed` for every single assertion. And if a test fails, it will not show all the other tests before/after it that passed/failed, it will only show the first failed test.
```

Comment:
It's not that assertions are "combined", it's rather that the test output panel presents the result of a test case, and not a result of an assertion. I am not even sure if I would mention it here, because (I think that) it's a normal behavior, expected from a sane testing framework, as opposed to log entry per assertion as it is for Python.

IMO this paragraph is useful for Python (and other languages which work in similar way), because there the test output is somewhat unusual, with one entry per assertion. I am not sure if I would put it for languages which behave in an expected way.